packages/lexical/src/LexicalReconciler.ts

```diff
@@ -284,7 +284,7 @@ function isLastChildLineBreakOrDecorator(
 ): boolean {
   const childKey = children[children.length - 1];
   const node = nodeMap.get(childKey);
-  return $isLineBreakNode(node) || $isDecoratorNode(node);
+  return $isCodeLineNode(node) || $isDecoratorNode(node);
```

Comment:
Need to rollback linebreak -> isColdeLineNode changes in reconciler, selection and utils. We should keep all changes for the code blocks with lexical/code. Reason for it: codeNode is not a core node, meaning it's not required for editor to have it registered to work. Hence all features/functionality should be isolated into its own package

We can look into reasons why this change was required after that